### ISSUE TITLE
csvlook is not handling encoding properly

### ISSUE THREAD

**Issue by zviri** [wireservice/csvkit#315]
Reading an utf-8 encoded file with csvlook fails on the well known "ascii' codec can't encode character" error. This happens because the strings that are being written to the output file are not encoded properly. I fixed the issue locally, so if it's welcomed I can provide the patch.

Using python 2.7.*

**Comment by vhcandido** [comment]
I still can reproduce this.
csvkit/cli.py : line 80
`self.output_file = sys.stdout`

**Comment by polvoazul** [comment]
@onyxfish I saw and used in many python projects the convention of having every command line option automatically available as an env variable  following the convention PROGRAMNAME_LONGOPTION for naming the envvar.
It worked quite well, was consistent and easy to document and implement :)

**Comment by jarib** [comment]
I'm still seeing this problem. Here's an example with some common norwegian non-ASCII characters:

```console
$ echo æåø | xxd
0000000: c3a6 c3a5 c3b8 0a                        .......
$ echo æåø | csvlook
|-------|
|  æåø  |
|-------|
$ echo æåø | csvlook | echo

'ascii' codec can't encode characters in position 3-5: ordinal not in range(128)
$ echo æåø | csvlook -e utf-8 | echo

'ascii' codec can't encode characters in position 3-5: ordinal not in range(128)
$ locale
LANG="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL=
$ pip freeze | grep csvkit
csvkit==0.9.1
$
```

**Comment by jpmckinney** [comment]
Install the latest csvkit:

    pip install -e git+git://github.com/onyxfish/csvkit.git@master#egg=csvkit

**Comment by fconil** [comment]
I'm having the same issue as reported above with non-ASCII characters where csvkit utilities output is not some csv data (with csvkit 0.9.1 and an utf-8 locale).

I tried to use the latest csvkit but the current version `ab176a5` does not work and fails on Travis. Even if there is no PyPI release, tagging a stable version could be useful.

```bash
$ cat csvkit-encoding-pb.csv
numéro,commentaire
1,"vous êtes français ?"
2,"La chèvre"
3,"Un rôle éprouvant"

$ csvlook csvkit-encoding-pb.csv
|---------+-----------------------|
|  numéro | commentaire           |
|---------+-----------------------|
|  1      | vous êtes français ?  |
|  2      | La chèvre             |
|  3      | Un rôle éprouvant     |
|---------+-----------------------|

$ csvlook csvkit-encoding-pb.csv | more
'ascii' codec can't encode character u'\xe9' in position 6: ordinal not in range(128)

$ csvcut -n csvkit-encoding-pb.csv
  1: numéro
  2: commentaire

$ csvcut -n csvkit-encoding-pb.csv | more
'ascii' codec can't encode character u'\xe9' in position 8: ordinal not in range(128)

$ csvcut -c 1,2 csvkit-encoding-pb.csv
numéro,commentaire
1,vous êtes français ?
2,La chèvre
3,Un rôle éprouvant

$ csvcut -c 1,2 csvkit-encoding-pb.csv | more
numéro,commentaire
1,vous êtes français ?
2,La chèvre
3,Un rôle éprouvant
```
So I tried to use @zviri fix in `CSVKitUtility` in cli.py (in csvkit 0.9.1)

```python
        if output_file is None:
            self.output_file = codecs.getwriter('utf-8')(sys.stdout, 'strict')
        else:
            self.output_file = output_file
```
but it breaks some csv output as @onyxfish explained above

```bash
$ csvcut -c 1,2 data/csvkit-encoding-pb.csv
Your file is not "utf-8" encoded. Please specify the correct encoding with the -e flag. Use the -v flag to see the complete error.
```

**Comment by jpmckinney** [comment]
@fconil The current version only fails on a very specific edge-case on Travis - it works on 2.7 otherwise. I can't reproduce any of this with the latest version.

**Comment by fconil** [comment]
Here's what I get with the current version on python 2.7.12 (Ubuntu 16/04) with the same test file.

```bash
$ csvcut -v -n csvkit-encoding-pb.csv | more
Traceback (most recent call last):
  File ".../PyEnv2.7.12/utilitaires-from-src/bin/csvcut", line 11, in <module>
    load_entry_point('csvkit', 'console_scripts', 'csvcut')()
  File ".../PyEnv2.7.12/utilitaires-from-src/src/csvkit/csvkit/utilities/csvcut.py", line 48, in launch_new_instance
    utility.main()
  File ".../PyEnv2.7.12/utilitaires-from-src/src/csvkit/csvkit/utilities/csvcut.py", line 32, in main
    self.print_column_names()
  File ".../PyEnv2.7.12/utilitaires-from-src/src/csvkit/csvkit/cli.py", line 330, in print_column_names
    output.write('%3i: %s\n' % (i, c))
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 8: ordinal not in range(128)
```

**Comment by jpmckinney** [comment]
Attempting to re-apply fix in #690
